File: client-web-api/ruby/cmrs/new_cmr.rb
#!/usr/bin/env ruby

$LOAD_PATH << File.dirname(__FILE__) + '/../lib'
require 'trisano-web-api-cmr.rb'

trisano = TriSanoWebApiCmr.new
options = trisano.parse_args(ARGV)

page = trisano.get("/cmrs/new")
form = page.form('new_morbidity_event')
form = trisano.populate_form(form)
result = trisano.submit(form, form['commit']) 

errors = result.search(".//div[@class = 'errorExplanation']")
errors.each { |e|
  error = e.search(".//li")
  error.each { |detail|
    raise detail.text.strip
    exit
  }
}

exit 0

File: client-web-api/php/cmrs/new_cmr.php
#!/usr/bin/env php

<?php
require_once('../lib/trisano-web-api-cmr.php');

$trisano = new TrisanoWebApiCmr;

$trisano->get("/cmrs/new");
$trisano->parse_args($argv);
$trisano->populate_form();
if ($trisano->browser->getTransportError()) {
  die($trisano->browser->getTransportError());
}
$page = $trisano->browser->clickSubmitByName('commit');

exit(0);

?>
